optimade/filtertransformers/mongo.py

```diff
@@ -172,30 +172,84 @@ def property_zip_addon(self, arg):
 
     def _recursive_expression_phrase(self, arg):
         """Helper function for parsing `expression_phrase`. Recursively sorts out
-        the correct precedence for `$and`, `$or` and `$nor`.
+        the correct precedence for `$not', '$and` and `$or`.
 
         """
-        if len(arg) == 1:
-            # without NOT
-            return arg[0]
 
-        # handle the case of {"$not": {"$or": [expr1, expr2]}} using {"$nor": [expr1, expr2]}.
-        if "$or" in arg[1]:
-            return {"$nor": self._recursive_expression_phrase([arg[1]["$or"]])}
+        def handle_not_and(arg):
+            # Handle the case of {"$not": {"$and": [expr1, expr2]}}
+            # which is equal to {"or": [{"$not": expr1}, {"$not": expr2}]}}
+            # We have to check for the special case in which the "and" was created by a previous NOT
+            # e.g NOT (NOT ({"a": {"$eq" : 6}})) -> NOT({$and:[{"a": {"$ne" : 6}},{"a": {"$ne": None}}]})
+
+            expr1 = arg["$and"][0]
+            expr2 = arg["$and"][1]
+            key = list(expr1.keys())[0]
+            if expr1.keys() == expr2.keys():
+                if expr1.get(key) == {"$ne": None}:
+                    return self._recursive_expression_phrase(["NOT", expr2])
+                elif expr2.get(key) == {"$ne": None}:
+                    return self._recursive_expression_phrase(["NOT", expr1])
+            return {
+                "$or": [
+                    self._recursive_expression_phrase(["NOT", subdict])
+                    for subdict in arg["$and"]
+                ]
+            }
+
+        def handle_not_or(arg):
+            # Handle the case of {"$not": {"$or": [expr1, expr2]}} using:
+            # {"$not": {"$or": [expr1, expr2]}} == {"$and": [(NOT, expr1), (NOT, expr2)]}}
+            # {"$nor": [expr1, expr2]} is not convenient because nor will also return documents where the field in expr1 or exp2 is missing.
```

Comment:
Is this still the case now the expressions themselves have checks for null values?

Reply:
I think we only create null checks for expressions with `"$ne"`.
An expression like `{"$nor": [{"a":{ "$eq": 6}}, {"b": {"$gte": 10}}]}` will also return documents for which a or b are `null`. 
So I think this is still the case.